tensorflow/core/common_runtime/device.cc

```diff
@@ -54,4 +54,22 @@ DeviceAttributes Device::BuildDeviceAttributes(
   return da;
 }
 
+
+bool Device::IsRemoteCallAllowed() const {
```

Comment:
Could you avoid the hardcode here?

Reply:
Seems straightforward with the exception of TPU_SYSTEM which is configured into VirtualDevice.

But this question led me to discovery of DeviceAttributes. So another approach seems to be to add is_remote_call_supported to DeviceAttribute message and implement IsRemoteCallSupported as a non-virtual attribute getter. Do you think that would be a better solution?

Reply:
To keep the change concise I eventually opted to keep base implementation of IsRemoteCallAllowed like this. Solving for VirtualDevice would snowball to extending DeviceAttributes. Keeping this allows to remove some overrides that you commented later on and provably keeps behavior of existing devices unchanged.